integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/VariantAnimationTest.kt

```diff
@@ -35,6 +35,10 @@ import com.android.designcompose.annotation.Design
 import com.android.designcompose.annotation.DesignComponent
 import com.android.designcompose.annotation.DesignDoc
 import com.android.designcompose.annotation.DesignVariant
+import com.android.designcompose.serdegen.Bezier
```

Comment:
I'm not sure if we should expose these types directly (partly because we're going to change to proto serialization, partly because they're currently clunky because serdegen makes ugly interfaces).

I know Compose has "AnimationSpec" which we could use instead of the serdegen curve definitions. I think we'd have to define a Kotlin type for whether it's smart animate or something else. Then if we wanted to do a very custom thing with different curves for different elements, or additional property changes that are only in the transition (like an extra rotation or blink or something), there are types to extend to do that.

Does that make sense, or it's better to use just the DCF types? Otherwise I like this change because it's pretty simple but can give a lot of options.

Reply:
Does "SmartAnimate" vs the other types of animation actually animate differently? It looked to me like they all did the same thing.

Reply:
Ok I updated so that the custom animation function returns a `AnimationTransition` interface that lets you access an `AnimationSpec<Float>`. Right now there is a `SmartAnimationTransition` that implements the interface. We could add more later but not sure that's the approach you were thinking of... and adding more transition implementations would require a little more plumbing.

Reply:
Yeah, I think this is good. Currently only "SmartAnimate" is implemented. Other animation types that Figma has are for the most part simpler because they are just simple translate + opacity done on the incoming and outgoing root views.